src/main/java/net/dv8tion/jda/api/events/interaction/commands/MessageContextCommandEvent.java

```diff
@@ -0,0 +1,41 @@
+package net.dv8tion.jda.api.events.interaction.commands;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageChannel;
+import net.dv8tion.jda.api.interactions.commands.interactions.MessageCommandInteraction;
+import net.dv8tion.jda.internal.interactions.commands.MessageCommandInteractionImpl;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Indicates that a message command was used in a {@link MessageChannel}.
+ *
+ * <h2>Requirements</h2>
+ * To receive these events, you must unset the <b>Interactions Endpoint URL</b> in your application dashboard.
+ * You can simply remove the URL for this endpoint in your settings at the <a href="https://discord.com/developers/applications" target="_blank">Discord Developers Portal</a>.
+ */
+public class MessageContextCommandEvent extends GenericCommandEvent implements MessageCommandInteraction
```

Comment:
Would it make more sense to just combine all context menu interactions into a single type that has individual (nonnull) getters:

- getTargetType / getCommandType
- getTargetMessage
- getTargetUser
- getTarget<...>

@DV8FromTheWorld what do you think? I think that would offer the same feature but require less work for extensions.